### ISSUE TITLE
"Type not found" in Windows Forms code

### ISSUE THREAD

**Issue by Alex527** [IronScheme/IronScheme#120]
Hi @leppie 

I'm new to Scheme and am currently playing with Windows Forms app I found online and cannot make it run. Code is the following.

`(import
  (rnrs)
  (ironscheme)
  (ironscheme clr)
)
 
(clr-reference System.Windows.Forms)
(clr-using System.Windows.Forms)

(define main-form (clr-new Form))
(clr-prop-set! Form Text main-form "--== Hello, world ==--")
(clr-prop-set! Form Size  main-form (clr-new System.Drawing.Size 640 480))

(define lbl-message (clr-new Label))
(clr-prop-set! Label Text lbl-message "Message: \"Hello, Windows Forms(IronScheme) World!\"")
(clr-prop-set! Label Size  lbl-message (clr-new System.Drawing.Size 320 20))
(clr-prop-set! Label Location  lbl-message (clr-new System.Drawing.Point 150 200))

(clr-call Form+ControlCollection Add (clr-prop-get Form Controls main-form) lbl-message)

(clr-static-call Application Run (clr-cast Form main-form))`

It fails on Form+ControlCollection Add line reporting the type not found.

`> (clr-call Form+ControlCollection Add (clr-prop-get Form Controls main-form) lbl-message)
Unhandled exception during evaluation:
&who: "ExtractTypeInfo"
&message: "type not found"
&syntax:
  form: "Form+ControlCollection"
  subform: #f`

But when I execute (clr-prop-get Form Controls main-form) it reports a proper type.

`> (clr-prop-get Form Controls main-form)
#<clr-type System.Windows.Forms.Form+ControlCollection>`

Am I doing something wrong? How to properly add controls to a form collection?

**Comment by leppie** [comment]
I think that is just the way .NET prints it.

Did you try `Form.ControlCollection` ? 

Basically it does `Type.GetType` for the lookup if you want to try in C#.

Let me know if it works, if not I will check what happens.

**Comment by Alex527** [comment]
Thanks for help!
After imporing `System.Collections` `IList Add` did the trick.
I imported `System.Collections.Generic` as well but `Form+ControlCollection` is still not found.
I tried `Forms.ControlCollection`, `ControlCollection` and `ArrangedElementCollection` (also a base class for `ControlCollection`) they are not found as well. I didn't try any types with generic parameters though.
I'm on version `IronScheme-1.0.266-a52b5aa` running `IronScheme.Console-v4.exe`.
